massa-execution-worker/src/execution.rs

```diff
@@ -196,6 +197,8 @@ impl ExecutionState {
             execution_trail_hash,
         )));
 
+        let cur_execution_version = execution_context.lock().execution_component_version;
```

Comment:
Instead of taking a lock here:
* create the bare execution context
* then get a copy of its execution component version
* then wrap it in arc mutex